utils/priority_queue/priority-queue.go

```diff
@@ -32,13 +33,13 @@ func (pq *wrapper[V, P]) Swap(i, j int) {
 	(*pq)[j].index = j
 }
 
-func (pq *wrapper[V, P]) Push(x *item[V, P]) {
-	item := x
+func (pq *wrapper[V, P]) Push(x any) {
```

Comment:
Why did you change it to any? If it only supports `(*Item[V, P])` and does not expect `item` to be `nil`, I think explicit type is better.

Reply:
Same as above